File: Swift/move-zeroes.swift
class Solution {
    func moveZeroes1(_ nums: inout [Int]) {
        var indexArray = [Int]()

        for (index, num) in nums.enumerated() {
            if num == 0 {
                indexArray.append(index)
            }
        }

        let zeroCount = indexArray.count
        guard zeroCount > 0 else { return }

        for _ in 1...zeroCount {
            let index = indexArray.popLast()!
            nums.remove(at: index)
            nums.append(0)
        }
    }

    func moveZeroes2(_ nums: inout [Int]) {
        var index = 0

        for num in nums {
            if num != 0 {
                nums[index] = num
                index += 1
            }
        }

        while index < nums.count {
            nums[index] = 0
            index += 1
        }
    }
}

File: Python/move-zeroes.py
class Solution:
    def moveZeroes1(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        indexArray = list()

        for index, num in enumerate(nums):
            if num == 0:
                indexArray.append(index)

        zeroCount = len(indexArray)
        if zeroCount <= 0:
            return

        for _ in range(zeroCount):
            index = indexArray.pop()
            nums.pop(index)
            nums.append(0)

    def moveZeroes2(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        index = 0

        for num in nums:
            if num != 0:
                nums[index] = num
                index += 1

        while index < len(nums):
            nums[index] = 0
            index += 1
